src/test/regress/sql/bfv_partition_plans.sql

```diff
@@ -1,6 +1,38 @@
 create schema bfv_partition_plans;
 set search_path=bfv_partition_plans;
 
+create table t1 (a int, b int, c text, d int) with (appendoptimized=true, orientation=column) distributed randomly partition by range(a) ( start (1) end (100) every (1), default partition extra );
+create table t2 (a int, b int, c text, d int) with (appendoptimized=true, orientation=column) distributed randomly partition by range(a) ( start (1) end (100) every (1), default partition extra );
+insert into t1 select  i%100, i, 'asdasdasdasasasdsdad'||i::text, i from generate_series(1, 1000)i;
+insert into t2 select * from t1;
+create or replace function get_executor_mem(query text) returns int language plpgsql as $$ declare
+    line text;
+    mem int[];
+    total int default 0;
+begin
+    for line in execute('EXPLAIN ANALYZE ' || query) loop
+        mem = regexp_matches(line, 'Executor memory: (\d+)K');
+        if mem is null then continue; end if;
+        total = total + mem[1];
+    end loop;
+    return total;
+end
+$$;
+with q as (
+    select $$select count(distinct a.d) from t1 a, t2 b where a.c = b.c and a.b > 50 and b.b > 50$$ as q
+), f as (
+    select  set_config('optimizer_enable_hashjoin', 'off', true),
+            set_config('optimizer_enable_materialize', 'off', true),
+            get_executor_mem(q.q) as m from q
+), n as (
+    select  set_config('optimizer_enable_hashjoin', 'on', true),
+            set_config('optimizer_enable_materialize', 'on', true),
+            get_executor_mem(q.q) as m from q
+) select abs(f.m - n.m) < 10000 from f, n;
+drop table t1;
+drop table t2;
+drop function get_executor_mem(query text);
+
```

Comment:
And add
`RESET client_min_messages;`
here.

Reply:
done